src/backend/distributed/planner/deparse_shard_query.c

```diff
@@ -520,6 +527,14 @@ GetTaskQueryType(Task *task)
 char *
 TaskQueryStringForAllPlacements(Task *task)
 {
+	if (GetTaskQueryType(task) == TASK_QUERY_NULL)
+	{
+		/* if task query type is TASK_QUERY_NULL then the data will be NULL,
+		 * this is unexpected state */
+		ereport(ERROR, (errcode(ERRCODE_INTERNAL_ERROR),
+						errmsg("unexpected task query state: task query type is null"),
+						errdetail("Please report this to the Citus core team.")));
+	}
```

Comment:
Minor readability note,  it looked easier to read this function.
```C
if (GetTaskQueryType(task) == TASK_QUERY_TEXT_LIST)
{
}
else if if (GetTaskQueryType(task) == TASK_QUERY_TEXT)
{
}
else if (GetTaskQueryType(task) == TASK_QUERY_NULL)
{

}
```
You can make the call on this.